### PR TITLE
Refactor `background.dart` in Dart Debug Extension

### PR DESCRIPTION
**Pull request by elliette**
This shouldn't affect or change any functionality. This is simply a refactor of the `background.dart` file to make it easier to read and understand.

The changes are:
1. Switch anonymous function callbacks to named functions so it's easier to understand what they are doing
2. Re-use some of the named functions in multiple places to reduce code duplication
3. Add lots of comments

### CODE REVIEW

File: dwds/debug_extension/web/background.dart [hunk 1]
```diff
@@ -149,84 +115,161 @@ void main() {
...
+}
+
+// Gets the current tab, then attaches the debugger to it:
+void _startDebugging(_) {
+  final getCurrentTabQuery = QueryInfo(active: true, currentWindow: true);
+
+  // Sends commands to debugger attached to the current tab.
+  // Extracts the extension backend port from the injected JS.
+  var attachDebuggerToTab = allowInterop((Tab currentTab) async {
+    if (!_debuggableTabs.contains(currentTab.id)) return;
```
Comment (annagrin) [inline dwds/debug_extension/web/background.dart:140]: Nit: can this be a separate function, similar to other refactoring that you have done for functions that are called via interop? I like that separation, makes it much easier to read the code?

File: dwds/debug_extension/web/background.dart [hunk 2]
```diff
@@ -149,84 +115,161 @@ void main() {
...
+
+void _handleMessageFromExternalExtensions(
+    Request request, Sender sender, Function sendResponse) async {
+  if (_allowedExtensions.contains(sender.id)) {
+    if (request.name == 'chrome.debugger.sendCommand') {
+      try {
+        var options = request.options as SendCommandOptions;
+        sendCommand(Debuggee(tabId: request.tabId), options.method,
+            options.commandParams, allowInterop(([e]) {
+          // No arguments indicate that an error occurred.
```
Comment (annagrin) [inline dwds/debug_extension/web/background.dart:226]: Same as above - looks like a helper.

**Review by annagrin — APPROVED** [review]
Thanks Elliott, great readability changes! Add some comments to possibly add more.

File: dwds/debug_extension/web/background.dart (hunk 1 above)
Comment (elliette) [reply dwds/debug_extension/web/background.dart]: Done!

File: dwds/debug_extension/web/background.dart (hunk 2 above)
Comment (elliette) [reply dwds/debug_extension/web/background.dart:228]: Done!

**Review by annagrin — APPROVED** [review]
Thanks you Elliott, this is much easier to read! LGTM!

File: dwds/debug_extension/web/background.dart (hunk 1 above)
Comment (annagrin) [reply dwds/debug_extension/web/background.dart]: THanks!

**Pull request merged**